src/pathwayvis/pathwayvis.component.ts

```diff
@@ -21,26 +21,53 @@ export class PathwayVisComponentController {
         this.$scope = $scope;
         this.mapOptions = mapOptions;
         this.showInfo = false;
+        this.$q = $q;
     }
 
-    public $onInit() {
-        this.mapOptions.init();
-
+    public async $onInit() {
         let item = this.$sharing.item('experiment');
         if (item) {
             this.mapOptions.addExpMapObject();
-            this.mapOptions.removeMapObject(0);
```

Comment:
You removed this because you are reassigning the objectMap with addRefMapObject right?